Sources/PayPal/PayPalClient.swift

```diff
@@ -37,31 +39,37 @@ public class PayPalClient {
     ///   - request: the PayPalRequest for the transaction
     ///   - presentingViewController: the ViewController to present PayPalPaysheet on, if not provided, the Paysheet will be presented on your top-most ViewController
     ///   - completion: Completion block to handle buyer's approval, cancellation, and error.
-    public func start(
-        request: PayPalRequest,
-        presentingViewController: UIViewController? = nil,
-        completion: @escaping (PayPalCheckoutResult) -> Void
-    ) {
+    public func start(request: PayPalRequest, presentingViewController: UIViewController? = nil) {
         CheckoutFlow.set(config: config, returnURL: returnURL)
 
         CheckoutFlow.start(
             presentingViewController: presentingViewController,
             createOrder: { order in
                 order.set(orderId: request.orderID)
             },
-            onApprove: { approval in
-                let payPalResult = PayPalResult(
-                    orderID: approval.ecToken,
-                    payerID: approval.payerID
-                )
-                completion(.success(result: payPalResult))
+            onApprove: { [weak self] approval in
+                self?.notifySuccess(for: approval)
             },
-            onCancel: {
-                completion(.cancellation)
+            onCancel: { [weak self] in
+                self?.notifyCancellation()
             },
-            onError: { errorInfo in
-                completion(.failure(error: PayPalError.nativeCheckoutSDKError(errorInfo)))
+            onError: { [weak self] errorInfo in
+                self?.notifyFailure(with: errorInfo)
```

Comment:
It looks like we are deallocating too quickly by using [weak self] in these closures and never hitting the delegate functions running this currently. Removing [weak self] allows us to hit the delegate functions as expected.

Reply:
It's less of a dealloc and more of a `no retain`. This way if the VC has been deallocated the optional `?` guards against a `nil` reference when the callback executes and it becomes a no-op.

Reply:
Right now, if you add a breakpoint to any of the `notify_` functions, they are not reliably being hit and therefore the delegate functions are never being called as the reference to self has already been deallocated in some cases.

Reply:
AH ok yeah that makes sense. Does checkout `nil` out those closures internally at some point?

Reply:
Yep, part of the shutdown steps as the paysheet closes is to call [`Checkout.resetForSession()`](https://github.paypal.com/nativexo/ios-native-checkout/blob/fdc6f913fabe600d8956dbeb6a0f0510c05e1260/Sources/PayPalCheckoutUI/Sources/Lifecycle%20and%20State/Shutdown%20Steps/ResetEnvironmentStep.swift#L27) which calls [`resetSessionCallbacks()`](https://github.paypal.com/nativexo/ios-native-checkout/blob/d2695f5f2ec886e559e4ac5cf2364de3253fd107/Sources/PayPalCheckoutUI/Sources/Lifecycle%20and%20State/Checkout.swift#L514) and sets the callbacks to `nil`